### PR TITLE
Remove P11 dependency on Ctypes

### PR DESCRIPTION
**Pull request by bbc2**
This mainly moves high level PKCS#11 bindings from `P11` to a new `P11_high_level` module.  This new module depends on `Pkcs11` but that's not the case of `P11` anymore.

### CODE REVIEW

File: doc/api.odocl [hunk 1]
```diff
@@ -77,6 +77,7 @@ Pkcs11_bindings
 Pkcs11_data
 Pkcs11_generator
 Pkcs11_helpers
+Pkcs11_high_level
```
Comment (emillon) [inline doc/api.odocl]: Can you rename it `P11_driver`? I think that it reflects better what this module does.

**Review by emillon — CHANGES_REQUESTED** [review]
That's almost exactly what I had in mind, thanks.

To make sure that everything works, can you restrict the `ctypes` dependency in `_tags` so that it is only used for `<src/pkcs11_*>` and `<src/p11_driver.*>`? I'll take care of splitting that into directories and making a subpackage in a separate PR.

File: doc/api.odocl (hunk 1 above)
Comment (bbc2) [reply doc/api.odocl]: Sure thing.  Done.

**Comment by bbc2** [comment]
And I just modified `_tags` as you suggested.

**Review by emillon — APPROVED** [review]

**Comment by emillon** [comment]
Thanks!

**Pull request merged**